machine/driver/driver.go

```diff
@@ -190,10 +215,46 @@ func (d *Driver) SetConfigFromFlags(opts drivers.DriverOptions) error {
 	d.RequestPayloads.InstanceCreateReq.AttachVPC = opts.StringSlice("vultr-vpc-ids")
 	d.RequestPayloads.InstanceCreateReq.SSHKeys = opts.StringSlice("vultr-ssh-key-ids")
 	d.RequestPayloads.InstanceCreateReq.DDOSProtection = utils.BoolPtr(opts.Bool("vultr-ddos-protection"))
-	d.RequestPayloads.InstanceCreateReq.UserData = opts.String("vultr-cloud-init-user-data")
 	d.RequestPayloads.InstanceCreateReq.ReservedIPv4 = opts.String("vultr-floating-ipv4-id")
 	d.RequestPayloads.InstanceCreateReq.ActivationEmail = utils.BoolPtr(opts.Bool("vultr-send-activation-email"))
 
+	cloudInitFromFile := opts.Bool("vultr-cloud-init-from-file")
+	cloudInitUserData := opts.String("vultr-cloud-init-user-data")
+	disableUFW := opts.Bool("vultr-disable-ufw")
+
+	if cloudInitFromFile {
+		data, err := os.ReadFile(cloudInitUserData)
+		if err != nil {
+			return fmt.Errorf("failed to read cloud-init file %q: %w", cloudInitUserData, err)
+		}
+
+		if disableUFW {
+			cloudConfigHeader := strings.TrimPrefix(string(data), "#cloud-config\n")
+			var config CloudConfig
+			if err := yaml.Unmarshal([]byte(cloudConfigHeader), &config); err != nil {
+				return fmt.Errorf("failed to unmarshal cloud config: %w", err)
+			}
+
+			config.RunCmd = append(config.RunCmd, `[ -x "$(command -v ufw)" ] && ufw disable`)
+
+			updatedCloudConfig, err := yaml.Marshal(&config)
+			if err != nil {
+				return fmt.Errorf("failed to marshal updated cloud-init data: %w", err)
+			}
+
+			newData := append([]byte("#cloud-config\n"), updatedCloudConfig...)
+			encodedUD := base64.StdEncoding.EncodeToString(newData)
+			d.RequestPayloads.InstanceCreateReq.UserData = encodedUD
+		} else {
+			encodedUD := base64.StdEncoding.EncodeToString(data)
+			d.RequestPayloads.InstanceCreateReq.UserData = encodedUD
+		}
+	} else {
+		if cloudInitUserData == "" {
+			cloudInitUserData = "I2Nsb3VkLWNvbmZpZwoKcnVuY21kOgogLSB1ZncgZGlzYWJsZQ=="
```

Comment:
Pass in the command check in this version, too `[ -x "$(command -v ufw)" ] && ufw disable`